### DISCUSSION TITLE
Best practice for user input text (i18n)

### DISCUSSION THREAD

**Discussion by mperrando** [buildo/bento-design-system#315] [Q&A] [answered]
I need to show inside Bento (e.g. a Label) components some text that is a user input and that will be not translated.

It is correct to pass such text through `unsafeLocalizedString()` or is there a better approach?

**Comment by gabro** [comment, marked as answer]
Hi @mperrando, it's correct, `LocalizedString` is there to ensure you don't forget localizing a component, but if you're sure this is user generated content you can definitely use `unsafeLocalizedString`.

Two more things:

- if these strings come from an external source, you may want to validate those and convert them to `LocalizedString` during the validation (e.g. using something like https://github.com/colinhacks/zod)

- I would recommend to limit the number of `unsafeLocalizedString` around the app, so if possible I would create a dedicated component like `UserGeneratedLabel` or something, which accepts `string` and wraps the actual `unsafeLocalizedString` usage
